LiteLoader/src/llapi/mc/PlayerAPI.cpp

```diff
@@ -1021,48 +1021,71 @@ bool Player::deletePlayerNbt(mce::UUID const& uuid) {
 
 std::unique_ptr<CompoundTag> getOfflineNbt(mce::UUID const& uuid) {
     auto serverId = getServerId(uuid);
-    if(serverId.empty()) {
-        return {};
+    if (serverId.empty()) {
+        return nullptr;
     }
-    if(!Global<DBStorage>->hasKey(serverId, playerCategory)) {
-        return {};
+    if (!Global<DBStorage>->hasKey(serverId, playerCategory)) {
+        return nullptr;
     }
     return Global<DBStorage>->getCompoundTag(serverId, playerCategory);
 }
 
+
+
 std::unique_ptr<CompoundTag> Player::getPlayerNbt(mce::UUID const& uuid) {
-    if(auto player = Global<Level>->getPlayer(uuid)) {
+    if (auto player = Global<Level>->getPlayer(uuid)) {
         return player->getNbt();
     }
     return getOfflineNbt(uuid);
 }
 
-bool Player::setPlayerNbt(mce::UUID const& uuid, CompoundTag& nbt) {
-    try{
+bool setOfflineNbt(mce::UUID const& uuid, CompoundTag* nbt) {
+    try {
+        auto &data = *nbt;
         auto serverId = getServerId(uuid);
-        if(serverId.empty()) {
+        if (serverId.empty()) {
             return false;
         }
-        Global<DBStorage>->saveData(serverId, nbt.toBinaryNBT(), playerCategory);
+        Global<DBStorage>->saveData(serverId, data.toBinaryNBT(), playerCategory);
         return true;
     }
     catch (const std::exception& exc) {
-        logger.error("Fail to set player nbt!\n{}",exc.what());
+        logger.error("Fail to set offline player nbt!\n{}",exc.what());
     }
     return false;
 }
 
-bool Player::setPlayerNbtTags(mce::UUID const& uuid, CompoundTag& data, vector<string> tags) {
-    try{
+bool Player::setPlayerNbt(mce::UUID const& uuid, CompoundTag* nbt) {
+    try {
+        auto serverId = getServerId(uuid);
+        if (serverId.empty()) {
+            return false;
+        }
+        bool res = true;
+        if (auto pl = Global<Level>->getPlayer(uuid)) {
+            return pl->setNbt(nbt);
+        }
+        else {
+            return setOfflineNbt(uuid, nbt);
+        }
+    }
+    catch (const std::exception& exc) {
+        logger.error("Fail to set player nbt!\n{}",exc.what());
+    }
+}
+
+bool Player::setPlayerNbtTags(mce::UUID const& uuid, CompoundTag* nbt, vector<string> tags) {
```

Comment:
```suggestion
bool Player::setPlayerNbtTags(mce::UUID const& uuid, CompoundTag* nbt, const vector<string>& tags) {
```
若函数内没有对tags进行赋值，推荐使用常量左值